internal/querynode/data_sync_service.go

```diff
@@ -36,257 +36,144 @@ const (
 
-	if _, ok := dsService.collectionFlowGraphs[collectionID]; !ok {
-		return nil, errors.New("collection flow graph doesn't existed, collectionID = " + fmt.Sprintln(collectionID))
-	}
-
-	tmpFGs := make(map[Channel]*queryNodeFlowGraph)
-	for _, channel := range vChannels {
-		if _, ok := dsService.collectionFlowGraphs[collectionID][channel]; ok {
-			tmpFGs[channel] = dsService.collectionFlowGraphs[collectionID][channel]
-		}
+	if _, ok := dsService.dmlFlowGraphs[channel]; !ok {
+		return nil, errors.New("DML flow graph doesn't existed, collectionID = " + fmt.Sprintln(collectionID))
 	}
 
-	return tmpFGs, nil
+	// TODO: return clone?
+	return dsService.dmlFlowGraphs[channel], nil
 }
 
-// getCollectionDeltaFlowGraphs returns the collection delta flowGraph by collectionID
-func (dsService *dataSyncService) getCollectionDeltaFlowGraphs(collectionID UniqueID, vChannels []string) (map[Channel]*queryNodeFlowGraph, error) {
+// getDeltaFlowGraph returns the delta flowGraph by collectionID
+func (dsService *dataSyncService) getDeltaFlowGraph(collectionID UniqueID, channel Channel) (*queryNodeFlowGraph, error) {
 	dsService.mu.Lock()
 	defer dsService.mu.Unlock()
 
-	if _, ok := dsService.collectionDeltaFlowGraphs[collectionID]; !ok {
-		return nil, errors.New("collection flow graph doesn't existed, collectionID = " + fmt.Sprintln(collectionID))
+	if _, ok := dsService.deltaFlowGraphs[channel]; !ok {
+		return nil, errors.New("delta flow graph doesn't existed, collectionID = " + fmt.Sprintln(collectionID))
 	}
 
-	tmpFGs := make(map[Channel]*queryNodeFlowGraph)
-	for _, channel := range vChannels {
-		if _, ok := dsService.collectionDeltaFlowGraphs[collectionID][channel]; ok {
-			tmpFGs[channel] = dsService.collectionDeltaFlowGraphs[collectionID][channel]
-		}
-	}
-
-	return tmpFGs, nil
+	// TODO: return clone?
+	return dsService.deltaFlowGraphs[channel], nil
 }
 
-// startCollectionFlowGraph starts the collection flow graph by collectionID
-func (dsService *dataSyncService) startCollectionFlowGraph(collectionID UniqueID, vChannels []string) error {
+// startDMLFlowGraph starts the DML flow graph by collectionID
+func (dsService *dataSyncService) startDMLFlowGraph(collectionID UniqueID, channel Channel) error {
 	dsService.mu.Lock()
 	defer dsService.mu.Unlock()
 
-	if _, ok := dsService.collectionFlowGraphs[collectionID]; !ok {
-		return errors.New("collection flow graph doesn't existed, collectionID = " + fmt.Sprintln(collectionID))
-	}
-	for _, channel := range vChannels {
-		if _, ok := dsService.collectionFlowGraphs[collectionID][channel]; ok {
-			// start flow graph
-			log.Debug("start collection flow graph", zap.Any("channel", channel))
-			dsService.collectionFlowGraphs[collectionID][channel].flowGraph.Start()
-		}
+	if _, ok := dsService.dmlFlowGraphs[channel]; !ok {
+		return errors.New("DML flow graph doesn't existed, collectionID = " + fmt.Sprintln(collectionID))
 	}
+	log.Debug("start DML flow graph", zap.Any("channel", channel))
```

Comment:
It's better to print out the collectionID

Reply:
fixed